bsdlib/doc/architecture.rst

```diff
@@ -0,0 +1,21 @@
+.. _architecture:
+
+Library architecture
+####################
+
+The purpose of the BSD library is to communicate with the nRF9160 modem firmware.
+This is done by sending events and getting responses over the Inter Processor Communication (IPC) peripheral.
+To facilitate data transfer between the application and the modem, payload data is set in the shared RAM.
+
+The modem contains a full IP and DTLS/TLS stack as well as GNSS.
+The library internally communicates over a Remote Procedure Call (RPC) protocol using the IPC and the shared RAM to utilize the modules inside the modem.
```

Comment:
What is a `module` here? Is would call it a `stack` e.g. `to utilize the protocol stacks inside the modem`.